pkgs/development/compilers/rust/rustc.nix

```diff
@@ -110,8 +110,13 @@ stdenv.mkDerivation {
   # back to darwin 10.4. This causes the OS name to be recorded as
   # "10.4" rather than the expected "osx". But mk/rt.mk expects the
   # built library name to have an "_osx" suffix on darwin.
+  # The `timeouts` and `ttl` tests in the tcp module are fragile and
+  # tend to fail on hydra, so we take off the `test` attribute.
   optionalString stdenv.isDarwin ''
     substituteInPlace mk/rt.mk --replace "_osx" "_10.4"
+    sed -z -e 's/#\[test\]\n[[:space:]]*fn timeouts()/#[cfg_attr(target_os = "macos", ignore)]\n    #[test]\n    fn timeouts()/' \
```

Comment:
The problem is that it will silently break as soon as the `sed` doesn't match anymore. At least with a `.patch` we will get a build error instead of sporadic failures because it doesn't apply cleanly anymore. Also note that there are already other tests we disable via this mechanism.

If you really don't want to create a patchfile, please use `substituteInPlace` instead of `sed` as it will be much more readable (as suggested in https://nixos.org/nixpkgs/manual/#ssec-stdenv-functions). Please note that there is no need to use the `cfg_attr(target_os)="macos"` as `sed` is only called for darwin anyway. Just disabling the test completely instead of conditionally should simplify the line even more.